georchestra-integration/externalized-accounts/src/main/java/org/geonetwork/security/external/integration/UserSynchronizer.java

```diff
@@ -103,13 +109,16 @@ public void synchronizeAll(List<CanonicalUser> canonical) {
         requireNonNull(canonical);
         canonical.forEach(u -> requireNonNull(u, "null references not accepted in user's list"));
 
-        log.debug("Synchronizing {} canonical user definitions...", canonical.size());
+        log.info("Synchronizing {} canonical user definitions...", canonical.size());
+        externalUserLinksList = new ArrayList<>();
+        i = 0;
         try {
             final Set<String> canonicalIds = canonical.stream().map(CanonicalUser::getId).collect(Collectors.toSet());
             final Map<String, UserLink> currentLinks = getExistingUserLinksById();
 
             deleteGoneUsers(canonicalIds, currentLinks);
             canonical.forEach(this::synchronize);
+            linksRepo.saveAll(externalUserLinksList);
```

Comment:
`externalUserLinksList` does not seem to have been modified since line 113 where the object is instantiated to an empty list, so I don't really get what is the purpose here ?